app/controllers/api/alert_definitions_controller.rb

```diff
@@ -31,5 +36,14 @@ def edit_resource(type, id = nil, data = {})
     def set_additional_attributes
       @additional_attributes = %w(expression)
     end
+
+    def expression_values(data)
+      data.slice("expression", "miq_expression", "hash_expression").values.compact
+    end
+
+    def create_miq_expression(data)
+      data["expression"] = MiqExpression.new(data["expression"]) if data["expression"]
```

Comment:
I'm not sure I understand why this is necessary. Type coercion should be handled by the setter in the model - if that's not currently the case then that needs to be added so we don't have to munge the data here (and anywhere else for that matter).

Reply:
OK, moved it to the model: https://github.com/ManageIQ/manageiq/pull/16397